src/components/layerselector/LayerSelectorDirective.js

```diff
@@ -0,0 +1,148 @@
+(function() {
+  goog.provide('ga_layerselector_directive');
+
+  goog.require('ga_map');
+  goog.require('ga_permalink');
+
+  var module = angular.module('ga_layerselector_directive', [
+    'ga_map',
+    'ga_permalink',
+    'pascalprecht.translate'
+  ]);
+
+  module.directive('gaLayerSelector',
+    function(gaPermalink, gaLayers, gaLayerFilters) {
+      return {
+        restrict: 'A',
+        templateUrl: 'components/layerselector/partials/layerselector.html',
+        scope: {
+          map: '=gaLayerSelectorMap'
+        },
+        link: function(scope, elt, attrs) {
+          scope.isClosed = true;
+          var map = scope.map;
+          var firstLayerChangeEvent = true;
+          var isOfflineToOnline = false;
+          scope.backgroundLayers = [
+            {id: 'ch.swisstopo.swissimage', label: 'bg_luftbild'},
+            {id: 'ch.swisstopo.pixelkarte-farbe', label: 'bg_pixel_color'},
+            {id: 'ch.swisstopo.pixelkarte-grau', label: 'bg_pixel_grey'},
+            {id: 'voidLayer', label: 'void_layer'}];
+
+          function setCurrentLayer(layerid) {
+            var layers = map.getLayers();
+            if (layerid == 'voidLayer') {
+              if (layers.getLength() > 0 &&
+                  layers.item(0).background === true) {
+                layers.removeAt(0);
+              }
+            } else if (gaLayers.getLayer(layerid)) {
+              var layer = gaLayers.getOlLayerById(layerid);
+              layer.background = true;
+              layer.displayInLayerManager = false;
+              if (layers[0] && layers[0].background) {
+                layers.setAt(0, layer);
+              } else {
+                layers.insertAt(0, layer);
+              }
+            }
+            gaPermalink.updateParams({bgLayer: layerid});
+            scope.currentLayer = layerid;
```

Comment:
You don't need to attach it to the scope as it is not used in the html of the directive.